### DISCUSSION TITLE
Parsing links to attached files

### DISCUSSION THREAD

**Discussion by jingxuanlim** [rusq/slackdump#70] [Q&A] [answered]
slackdump is incredibly helpful to me. Many thanks for building this utility!

I have a question about parsing the `.json` output and the example python code helped me greatly. Currently, a http link is recorded for attached files. Is there a way of automatically downloading attached files (such as images, videos and text files) so that they also display nicely in the python output?

I also wonder if there is an option, since attachments are already downloaded into a folder, to have the links point to those files instead?

**Comment by rusq** [comment, marked as answer]
Hey, @jingxlim thanks for the kind words and feedback, I'm glad that you're finding slackdump helpful.

Currently slackdump does not update the file link URLs, just downloads them.  I was planning to update this part for a while now, as file attachments are not quite useful in their current form rather than for "archival" purposes, so expect it soon.

**Reply by jingxuanlim** [reply]
Sounds good. Looking forward to it!

**Comment by pican79** [comment]
Hi @rusq,
can you elaborate on what you plan to change exactly?

Currently, the download path of each file (relative to the json/text dump file) seems to be a concatenation of:
- root object `channel_id` property
- `/` character
- file object `id` property
- `-` character
- file object `name` property

**Reply by rusq** [reply]
That's correct, the path will remain the same - the message[i].File[j].PrivateURL field for each file will be updated with the local path to the file, relative to the JSON file.  Right now it's a URL to original file location on slack server
